src/vaadin-grid-styling-mixin.html

```diff
@@ -0,0 +1,73 @@
+<!--
+@license
+Copyright (c) 2018 Vaadin Ltd.
+This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+-->
+<script>
+  window.Vaadin = window.Vaadin || {};
+  window.Vaadin.Grid = window.Vaadin.Grid || {};
+
+  /**
+   * @polymerMixin
+   */
+  Vaadin.Grid.StylingMixin = superClass => class StylingMixin extends superClass {
+
+    static get properties() {
+      return {
+        /**
+         * A function that allows generating CSS class names for grid cells
+         * based on their row and column. The return value should be the generated
+         * class name as a string, or multiple class names separated by whitespace
+         * characters.
+         *
+         * Receives two arguments:
+         * - `rowData` The object with the properties related with
+         *   the rendered item, contains:
+         *   - `rowData.index` The index of the item.
+         *   - `rowData.item` The item.
+         *   - `rowData.expanded` Sublevel toggle state.
+         *   - `rowData.level` Level of the tree represented with a horizontal offset of the toggle button.
+         *   - `rowData.selected` Selected state.
+         * - `column` The `<vaadin-grid-column>` element.
+         */
+        cellClassNameGenerator: Function
+      };
+    }
+
+    static get observers() {
+      return [
+        '__cellClassNameGeneratorChanged(cellClassNameGenerator)'
+      ];
+    }
+
+    __cellClassNameGeneratorChanged(cellClassGenerator) {
+      this.generateClassNames();
+    }
+
+    /**
+     * Runs the `cellClassNameGenerator` for the visible cells.
+     * If the generator depends on varying conditions, you need to
+     * call this function manually in order to update the styles when
+     * the conditions change.
+     */
+    generateClassNames() {
+      Array.from(this.$.items.children).forEach(
+        row => this._generateClassNames(row, this.__getRowModel(row)));
+    }
+
+    _generateClassNames(row, rowData) {
+      Array.from(row.children).forEach(cell => {
+        if (cell.__generatedClasses) {
```

Comment:
Inspecting the DOM, looks like the `td` -elements have no default class names at all. Could this loop be simplified to just
```
cell.className = '';
this.cellClassNameGenerator && (cell.className = this.cellClassNameGenerator(rowData, cell._column));
```

Reply:
On polyfilled browsers some classes are added to the `td`-elements, so we can't just clear them.

Reply:
Right, forgot about the shady css classes.